connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/InboundConnectorContextImpl.java

```diff
@@ -94,14 +99,28 @@ public InboundConnectorDefinition getDefinition() {
 
   @Override
   public void reportHealth(Health health) {
-    this.health = health;
+    if (this.health == null) {
+      this.health = health;
+      return;
+    }
+    this.health = this.health.merge(health);
   }
 
   @Override
   public Health getHealth() {
     return health;
   }
 
+  @Override
+  public void log(ActivityLog log) {
+    this.logs.add(log);
+  }
+
+  @Override
+  public Queue<ActivityLog> getLogs() {
+    return this.logs;
+  }
```

Comment:
Same with `getHealth()` should we really expose it to the clients?

Reply:
I agree with your thinking, let's keep the interface as slim as possible. But how can we expose this to ourselves without creating unnecessary complexity?

Reply:
We could add a `InboundConnectorReportingContext` interface to get access to this data points but keep it out of the SDK for example.